add morphos server

danvergara opened the pull request:
# App Submission

### App name
Morphos Server

### 256x256 SVG icon


[Morphos icon](https://svgshare.com/s/14x5)

### Gallery images
_(Upload 3 to 5 high-quality gallery images (1440x900px) of your app in PNG format, or just upload 3 to 5 screenshots of your app and we'll help you design the gallery images.)_

![morphos](https://github.com/getumbrel/umbrel-apps/assets/12239167/99a31104-5dd6-4b21-9d5b-4d815659a7e4)
![select_options_morphos](https://github.com/getumbrel/umbrel-apps/assets/12239167/bf0bae8b-91f1-4855-92dc-acfe30def803)
![modal_morphos](https://github.com/getumbrel/umbrel-apps/assets/12239167/6fa75713-01e8-442c-be1f-06f029d0755d)



### I have tested my app on:
- [ ] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

nmfretz commented on morphos/docker-compose.yml:
```diff
@@ -0,0 +1,16 @@
+      APP_PORT: 8080
+
+  web:
+    image: ghcr.io/danvergara/morphos-server:0.2.0
```
We need to add the multi-arch image digest here to make sure all installs and app restarts result in the same image being downloaded.

Since you are using ghcr.io and have a multi-arch build for your image you can find the right digest listed under the tag. Here's the digest for your latest 0.3.0 for example:

<img width="713" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/7c3cfbf8-8d8b-4d5c-803a-c49361c7cd3a">

So for 0.3.0 you would use:
```
ghcr.io/danvergara/morphos-server:0.3.0@sha256:b09a886b6ae1c2c3ab3911d0b3513b1a8c9953e80edeb942d33814700ebadfc7
```

danvergara replied:
I just replaced the old one with the one you suggest.
Thanks!

nmfretz commented on morphos/docker-compose.yml:
```diff
@@ -0,0 +1,16 @@
+    image: ghcr.io/danvergara/morphos-server:0.2.0
+    restart: on-failure
+    stop_grace_period: 1m
+    ports:
+      - 8080:8080
```
We don't need to expose a port on the host here. Instead, the app_proxy container + docker's internal dns handles this. You can delete the entire ports directive.

danvergara replied:
Gotcha!
It's gone!

nmfretz commented on morphos/docker-compose.yml:
```diff
@@ -0,0 +1,16 @@
+    stop_grace_period: 1m
+    ports:
+      - 8080:8080
+    volumes:
+      - ${APP_DATA_DIR}/tmp:/tmp
```
Do you need to persist the `tmp` directory on the host machine between app restarts? If not you can just remove this entire directive.

If you do need the files in `tmp` to persist across app restarts then leave this as-is, but we'll beed to commit this directory to the app repo so that it gets installed with correct permissions on a users Umbrel. If we don't do this then Docker will create the directories with root permissions. What you can do is create a `data` directory in the morphos app directory, and then inside the `data` directory create a `tmp` directory with a `.gitkeep` file in it to make sure it is committed even though it is an empty directory.

Take a look at Gitea as an example: https://github.com/getumbrel/umbrel-apps/tree/master/gitea

nmfretz commented on morphos/umbrel-app.yml:
```diff
@@ -0,0 +1,27 @@
+id: morphos
+category: files
+name: Morphos Server
+version: "0.1.1"
```
This can be updated to the current version of the app

nmfretz commented on morphos/umbrel-app.yml:
```diff
@@ -0,0 +1,27 @@
+category: files
+name: Morphos Server
+version: "0.1.1"
+tagline: Self-hosted file converter server
```
Up to you here, but feel free to remove "server" from the tag so that the average user just understands the app as a self-hosted file converter

nmfretz commented on morphos/umbrel-app.yml:
```diff
@@ -0,0 +1,27 @@
+name: Morphos Server
+version: "0.1.1"
+tagline: Self-hosted file converter server
+description: >-
+  Today we are forced to rely on third party services to convert files to other formats. This is a serious threat to our privacy, if we use such services to convert files with  highly sensitive personal data. It can be used against us, sooner or later.
+
+  Morphos server aims to solve the problem mentioned above, by providing a self-hosted server to convert files privately. The project provides an user-friendly web UI.
+
+  For now, Morphos only supports images. Documents will be added soon.
```
Rendering the yaml nicely is a bit tricky. To get a blank line for whitespace between paragraphs you need to add **2** blank lines in the yaml.

Otherwise it looks like this:

<img width="713" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/cb723805-4177-43cf-be3d-71185abb026e">

danvergara replied:
I just rephrased the half of the description added a new list of feature, and of course, extra lines for better rendering.

nmfretz commented on morphos/umbrel-app.yml:
```diff
@@ -0,0 +1,27 @@
+  Morphos server aims to solve the problem mentioned above, by providing a self-hosted server to convert files privately. The project provides an user-friendly web UI.
+
+  For now, Morphos only supports images. Documents will be added soon.
+releaseNotes: >-
+  PDF support is finally here! Easily convert to images to PDF files.  Otherwise is also possible.
```
For the initial release on umbrelOS, we can leave the release notes blank.

```suggestion
releaseNotes: ""
```

danvergara replied:
Gone, too!

nmfretz commented on morphos/umbrel-app.yml:
```diff
@@ -0,0 +1,27 @@
+  For now, Morphos only supports images. Documents will be added soon.
+releaseNotes: >-
+  PDF support is finally here! Easily convert to images to PDF files.  Otherwise is also possible.
+developer: Dan Vergara
```
The other two pieces of info to add here are:

```yaml
submitter: your-name-or-handle-here
submission: https://github.com/getumbrel/umbrel-apps/pull/1020

danvergara replied:
Nice! I just added those missing fields.
Really appreciated!

nmfretz reviewed (commented):
Thanks for this converter app @danvergara! It looks great. I have left a review below with some packaging items to address. Once these are done I'll do a final test before sending this to the app store!

nmfretz commented:
Also @danvergara do you have any ideas for text that you'd like to see on each of the gallery images?

danvergara commented:
Thanks @nmfretz for taking the time to review this PR!

I just addressed all your comments. Thanks!

Regarding the text for the gallery I'd like: Convert your files privately.

nmfretz commented:
Excellent, thanks @danvergara this is working great. We'll get the gallery images and icon ready and then launch on the app store! Give us just a few days to do this due to our current workload.

danvergara commented:
No rush!

Looking forward to the launch on the app store!!

nmfretz commented:
New gallery assets are ready @danvergara! Thanks again for the app submission, it is now live.

<img width="1468" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/ebaf2743-a77c-4774-b5cd-5551b93153e4">

danvergara commented:
Sorry, I got busy over the past week.

I was able to download and use Morphos in my home lab running Umbrel. I was great!!!

Thank you so much @nmfretz!!
